gp-cli/src/main/java/com/ibm/g11n/pipeline/tools/cli/UpdateBundleCmd.java

```diff
@@ -34,34 +35,67 @@
 
     @Parameter(
             names = {"-l", "--languages"},
-            description = "List of bundle's target language ID(s) separted by comma.")
+            description = "List of bundle's target language ID(s) separted by comma. "
+                        + "Empty list \"\" will remove all existing target languages.")
     private String languageIdsListStr;
 
     @Parameter(
             names = {"-n", "--note"},
-            description = "Translation instruction note")
+            description = "Translation instruction note. "
+                        + "Empty note \"\" will remove existing instruction note.")
     private String note;
 
+    @Parameter(
+            names = {"-r", "--readOnly"},
+            description = "true to set the bundle read only.")
+    private String readOnlyStr;
+
     @Override
     protected void _execute() {
         Set<String> trgLangs = null;
         if (languageIdsListStr != null) {
             String[] langs = languageIdsListStr.split(",");
             trgLangs = new HashSet<>(langs.length);
             for (String lang : langs) {
+                if (lang.isEmpty()) {
+                    continue;
+                }
                 trgLangs.add(lang);
             }
         }
 
-        if (languageIdsListStr == null && note == null) {
+        List<String> notes = null;
+        if (note != null) {
+            if (note.isEmpty()) {
+                // Empty note will delete the existing note
+                notes = Collections.emptyList();
+            } else {
+                notes = Collections.singletonList(note);
+            }
+        }
+
+        Boolean readOnly = null;
+        if (readOnlyStr != null) {
+            if (readOnlyStr.equalsIgnoreCase("true")) {
+                readOnly = Boolean.TRUE;
+            } else if (readOnlyStr.equalsIgnoreCase("false")) {
+                readOnly = Boolean.FALSE;
+            } else {
+                System.out.println("Bad -r (--readOnly) argument value: " + readOnlyStr
+                        + ". The command argument will be ignored.");
```

Comment:
I think "--readOnly bogus" should produce warning message (or even error) and do not make any changes, instead of changing the state.